openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,690 @@
+import json
+import re
+import sys
+from collections import defaultdict
+from typing import Any, Callable, Dict, Iterable, List, Optional, Set, Tuple, Type, cast
+
+import fastjsonschema
+
+from openslides_backend.models.base import model_registry
+from openslides_backend.models.fields import (
+    BaseRelationField,
+    BaseTemplateField,
+    BooleanField,
+    CharArrayField,
+    CharField,
+    ColorField,
+    DecimalField,
+    Field,
+    FloatField,
+    GenericRelationField,
+    GenericRelationListField,
+    HTMLPermissiveField,
+    HTMLStrictField,
+    IntegerField,
+    JSONField,
+    NumberArrayField,
+    RelationField,
+    RelationListField,
+    TimestampField,
+)
+from openslides_backend.models.models import Model
+from openslides_backend.shared.patterns import Collection
+
+SCHEMA = fastjsonschema.compile(
+    {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "title": "Schema for initial and example data.",
+        "type": "object",
+        "patternProperties": {
+            "^[a-z_]+$": {
+                "type": "array",
+                "items": {
+                    "type": "object",
+                    "properties": {"id": {"type": "number"}},
+                    "required": ["id"],
+                },
+            }
+        },
+        "additionalProperties": False,
+    }
+)
+
+
+class CheckException(Exception):
+    pass
+
+
+def check_string(value: Any) -> bool:
+    return value is None or isinstance(value, str)
+
+
+color_regex = re.compile("^#[0-9a-f]{6}$")
+
+
+def check_color(value: Any) -> bool:
+    return value is None or bool(isinstance(value, str) and color_regex.match(value))
+
+
+def check_number(value: Any) -> bool:
+    return value is None or isinstance(value, int)
```

Comment:
This is not up-to-date: https://github.com/OpenSlides/OpenSlides/blob/77f6516438c93e790fd2915a32d416bb3a0f88a5/docs/datavalidator/check_json.py#L53

Reply:
Updated.